engine.js

```diff
@@ -193,6 +197,133 @@ var accumulateResult = function(ifResult, thenResult) {
         DEBUG = level;
     };
 
+    HMDAEngine.getDebug = function() {
+        return DEBUG;
+    };
+
+    /*
+     * -----------------------------------------------------
+     * Local DB
+     * -----------------------------------------------------
+     */
+
+    HMDAEngine.setUseLocalDB = function(bool) {
+        _USE_LOCAL_DB = bool;
+        if (bool) {
+            return resetDB();
+        } else {
+            return destroyDB();
+        }
+    };
+
+    HMDAEngine.shouldUseLocalDB = function() {
+        return _USE_LOCAL_DB;
+    };
+
+    HMDAEngine.loadCensusData = function(year) {
+        var apiCalls = [
+            this.apiGET('localdb/census/msaCodes'),
+            this.apiGET('localdb/census/stateCounty'),
+            this.apiGET('localdb/census/stateCountyMSA'),
+            this.apiGET('localdb/census/stateCountyTract'),
+            this.apiGET('localdb/census/stateCountyTractMSA')
+        ];
+        return Promise.map(apiCalls, function(body) {
+            return loadDB(resultFromResponse(body));
+        });
+    };
+
+    var resetDB = function() {
+        return destroyDB()
+        .then(function() {
+            _LOCAL_DB = levelup('hmda');
+            return _LOCAL_DB;
+        });
+    };
+
+    var destroyDB = function() {
+        var deferred = Promise.defer();
+
+        var realDestroy = function() {
+            if (typeof levelup.destroy === 'function') {
+                levelup.destroy('hmda', function(err) {
+                    if (err) {
+                        deferred.reject(err);
+                    }
+                    _LOCAL_DB = null;
+                    deferred.resolve();
+                });
+            } else {
+                var request = indexedDB.deleteDatabase('IDBWrapper-hmda');
+                request.onsuccess = function() {
+                    _LOCAL_DB = null;
+                    deferred.resolve();
+                };
+                request.onerror = function(err) {
+                    deferred.reject(err);
+                };
+            }
+        };
+        if (_LOCAL_DB) {
+            _LOCAL_DB.close(function(err) {
+                if (err) {
+                    deferred.reject(err);
+                }
+                realDestroy();
+            });
+        } else {
+            realDestroy();
+        }
+        return deferred.promise;
+    };
+
+    var loadDB = function(data) {
+        var deferred = Promise.defer();
+        _LOCAL_DB.batch(data, function(err) {
+            if (err) {
+                deferred.reject(err);
+            }
+            deferred.resolve();
+        });
+        return deferred.promise;
+    };
+
+    var isValidCensusCombination = function(censusparams) {
```

Comment:
For consistency, can you either prefix this with `local` like `localMSALookup()` or switch `localMSALookup()` to `getMSAName()`?